ARTICLE.md

```diff
@@ -1,27 +1,290 @@
+    version: v1
+---
+kind: Route
+apiVersion: v1
+metadata:
+  name: my-route
+  namespace: my-app
+  labels:
+    app: my-app
+    component: my-component
+spec:
+  host: my-component.apps.example.com
+  to:
+    name: my-component-v1
+    kind: Service
+  port:
+    targetPort: http
+```
+
+Even better, we can then secure our web service with edged
+terminated TLS and the ensure our users are always redirected
+to the HTTPS port of the load-balancer:
+
+```
+  tls:
+    termination: edge
+    insecureEdgeTerminationPolicy: Redirect
+```
+
+so now our `Route` as a whole looks like:
+
+```
+---
+kind: Route
+apiVersion: v1
+metadata:
+  name: my-route
+  namespace: my-app
+  labels:
+    app: my-app
+    component: my-component
+spec:
+  host: my-component.apps.example.com
+  to:
+    name: my-component-v1
+    kind: Service
+  port:
+    targetPort: http
+  tls:
+    termination: edge
+    insecureEdgeTerminationPolicy: Redirect
+```
+
+#### End-to-End TLS Made Easier
+
+Alright so that was pretty easy and we are a few steps closer
+to achieving end-to-end TLS. Moving forward, we first need our
+`Route` to no longer edge terminate TLS, which leaves us with
+two options for the termination policy: `passthrough` and
+`reencrypt`.
+
+With `passthrough`, the load-balancer will not participate in
+the TLS handshake and will simply send all traffic straight to
+the `Service` and then your `Pod`'s. That means your `Pod`'s
+would need a valid TLS cert for `my-component.apps.example.com`
+for the sake of the external client.
+
+With `reencrypt`, the load-balancer will handle the TLS
+handshake with the external client. Then it will do its own TLS
+handshake with the `Pod`'s at the end of your `Service`,
+re-encrypting the packets based on the TLS cert your pods are
+serving. Depending on what cert your `Pod`'s serve, you can
+specify a `destinationCACertificate` in the `Route` so that it
+trusts your cert. Run `oc explain route.spec.tls` for more
+details on all three policies and what additional options you
+have.
+
+Aside from needing a valid cert
+`my-component.apps.example.com`, the other drawback to
+`passthrough` is that internal `Service` hostname, i.e.
+`my-component-v1.my-app.svc.cluster.local`, would still be
+considered invalid. With `reencrypt` you could still achieve
+this, but then your `Pod`'s need a valid cert for that internal
+hostname... luckily OpenShift can automatically provide just
+that!
+
+OpenShift has a [special `Service` annotation]() that will tell
+it to automatically create a cert, all you add is the following:
+
+```
+annotations:
+    service.alpha.openshift.io/serving-cert-secret-name: my-component-v1-tls
+```
+
+OpenShift will then make a [`kubernetes.io/tls`]() type
+`Secret` called `my-component-v1`, and it will contain the key
+and cert in PEM format. We will discuss ways you can have your
+application `Pod`'s consume this `Secret` in the next article.
+
+But so, putting this altogether our setup now looks like:
```

Comment:
Haha I hear your voice saying this, but I'd remove the `But so,`. It sounds fine talking, but doesn't read great.